src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -731,7 +731,8 @@ int getSlotForFieldOrProperty(Symbol member)
 
         private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
         {
-            if (_symbol is MethodSymbol method &&
+            // DoesNotReturn is only supported in member methods
```

Comment:
>DoesNotReturn [](start = 15, length = 13)

Can `[DoesNotReturn]` be applied to a local function?

Reply:
We don't complain if it is applied, but we do not change analysis when it is present.

I actually expected us to mark regions unreachable in NullableWalker when calling a local function that does not return, and thus give no warnings in those regions--definite assignment already works such that the following scenario gives no errors. Unfortunately this does not seem to be the case. It's an inconsistency we might want to follow up on.

```cs
string s;
local1();
s.ToString();

void local1() => throw null!;
```

```cs
string? s = null;
local1();
s.ToString(); // this warns, but shouldn't

void local1() => throw null!;
```

Reply:
I will try and ensure that DoesNotReturn does not impact analysis of local functions for the moment, and file a follow-up issue to confirm that this is expected.